src/flp/types/fixedpoint_l2.rs

```diff
@@ -523,6 +558,51 @@ where
         Ok(decoded_vector)
     }
 
+    fn add_noise(&self, aggregate_share: &mut Vec<F>) -> Result<(), FlpError> {
+        let get_noise = || -> Result<F::Integer, FlpError> {
+            // we get the noise as bigint, so we have to convert it to i128,
+            // for this we compute its modulo wrt the field modulus, then get
+            // the i128 value, which we put into the field.
+
+            // 1. get noise
+            let (ref a, ref b) = self.noise_parameter;
+            let noise: BigInt =
+                sample_discrete_gaussian(a, b).map_err(|e| FlpError::Noise(e.to_string()))?;
+
+            // 2. noise as i128
+            let noise: BigInt = noise % BigInt::from(F::modulus());
```

Comment:
very good question. our reasoning is that for a field element `f` interpreted as an integer, and integer noise `n`, it is equivalent to do any of the following:
- `(f + n%m)%m` which is what we do here
- `(f+n)%m` which means adding the actual noise and then projecting back to the field as post-processing.
the second gives DP because it's using discrete gaussian noise on an integer-valued thing and that's invariant under post-processing.

Reply:
Bump (this conversation is a bit stale but it sounds like it may be worth documenting the invariant in a comment?)